gpu_ocean/SWESimulators/IEWPFOcean.py

```diff
@@ -554,25 +614,69 @@ def _createS(self, ensemble):
                     tmp_y[b, a] += y_corr[j,i]*self._SOAR_Q_CPU(a, b, i, j)
         if self.debug: self.showMatrices(tmp_x, tmp_y, "$Q_{SOAR} U_{GB}^T H^T$")   
         
+        
         # Apply the SOARfunction again to fill the points needed to find drift in (mid_i, mid_j)
-        # For both x and y:
-        # This means that we only need to evaluate Q_{SOAR} Q_{SOAR} U_{GB}^T H^T at four points
-        for j,i in [mid_j+1, mid_i], [mid_j-1, mid_i], [mid_j, mid_i-1], [mid_j, mid_i+1]:
-            x_corr[j,i] = 0
-            y_corr[j,i] = 0
-            for b in range(j-2, j+3):
-                for a in range(i-2, i+3):
-                    SOAR_Q_res = self._SOAR_Q_CPU(a, b, i, j)
-                    x_corr[j,i] += tmp_x[b, a]*SOAR_Q_res
-                    y_corr[j,i] += tmp_y[b, a]*SOAR_Q_res
-            if self.debug: print ("(j, i ,x_corr[j,i], y_corr[j,i]): ", (j, i ,x_corr[j,i], y_corr[j,i]))
+        # In order to reuse the bicubic interpolation routine, we need to calculate the SOAR values
+        # in the 5x5 area centered in the mid-cell.
+        # The values outside of the 7x7 buffers are all zero, so we don't have to loop over them.
+        for j in range(mid_j-2, mid_j+3):
+            for i in range(mid_i-2, mid_i+3):
+                x_corr[j,i] = 0
+                y_corr[j,i] = 0
+                for b in range(max(j-2,0), min(j+3,7)):
+                    for a in range(max(i-2,0), min(i+3,7)):
+                        SOAR_Q_res = self._SOAR_Q_CPU(a, b, i, j)
+                        x_corr[j,i] += tmp_x[b, a]*SOAR_Q_res
+                        y_corr[j,i] += tmp_y[b, a]*SOAR_Q_res
+                if self.debug: print ("(j, i ,x_corr[j,i], y_corr[j,i]): ", (j, i ,x_corr[j,i], y_corr[j,i]))
         if self.debug: self.showMatrices(x_corr, y_corr, "$Q_{SOAR} Q_{SOAR} U_{GB}^T H^T$")
-
+        
+        
+        # Obtaining the values required for finding geostrophic balanced values in the center cell.
+        # Need only consider one cell in each direction.
+        # We find these values by interpolation or direct reading of x_corr and y_corr values
+        if self.interpolation_factor > 1:
+            
+            # Use a function alias to get more readable code
+            # This function has input values: coarse_eta, coarse_i, coarse_j, rel_x, rel_y
+            interpolation_alias = ensemble.particles[0].small_scale_model_error._bicubic_interpolation_inner
+            
+            # Relative offset of neighbouring fine grid point
+            rel_offset = 1.0/self.interpolation_factor
+            
+            # north and east values are calculated from the surface in the north-east direction
+            # south and west values are calculated from the surface in the south-west direction 
+            
+            x_north = interpolation_alias(x_corr, mid_i  , mid_j  ,          0.0,      rel_offset)
```

Comment:
Please add whitespace only after commas, not before :)

(Will not comment this in the rest of the code.)

Reply:
As I see it, the only reason to use more than one space in places like this, is to align parameters across multiple lines. For me, it is then also natural to align the commas, and I don't really see why that should be a problem :)

Reply:
Parameters can be aligned with whitespace after commas (or other punctuation). I think whitespace _before_ commas makes the code harder to read. I also think this is supported in most standardized code styles. I will merge this PR, but please consider not using whitespace _before_ commas :)